src/code/z_play.c

```diff
@@ -55,38 +55,38 @@ void func_800BC5E0(GlobalContext* globalCtx, s32 transitionType) {
     transitionCtx->transitionType = transitionType;
 
     if ((transitionCtx->transitionType >> 5) == 1) {
-        transitionCtx->init = TransitionCircle_Init;
-        transitionCtx->destroy = TransitionCircle_Destroy;
-        transitionCtx->start = TransitionCircle_Start;
-        transitionCtx->isDone = TransitionCircle_IsDone;
-        transitionCtx->draw = TransitionCircle_Draw;
-        transitionCtx->update = TransitionCircle_Update;
-        transitionCtx->setType = TransitionCircle_SetType;
-        transitionCtx->setColor = TransitionCircle_SetColor;
-        transitionCtx->setEnvColor = TransitionCircle_SetEnvColor;
+        transitionCtx->init = (TransitionInit)TransitionCircle_Init;
+        transitionCtx->destroy = (TransitionDestroy)TransitionCircle_Destroy;
+        transitionCtx->start = (TransitionStart)TransitionCircle_Start;
+        transitionCtx->isDone = (TransitionIsDone)TransitionCircle_IsDone;
+        transitionCtx->draw = (TransitionDraw)TransitionCircle_Draw;
+        transitionCtx->update = (TransitionUpdate)TransitionCircle_Update;
+        transitionCtx->setType = (TransitionSetType)TransitionCircle_SetType;
+        transitionCtx->setColor = (TransitionSetColor)TransitionCircle_SetColor;
+        transitionCtx->setEnvColor = (TransitionSetEnvColor)TransitionCircle_SetEnvColor;
```

Comment:
Why are these casts needed? Shouldn't the function types be the same as the types in the struct?
In general I'd say we should strongly avoid casting functions, so we should look for an alternative if possible